src/dynamics/spectral_grid.jl

```diff
@@ -44,14 +47,17 @@ $(TYPEDFIELDS)
     "[OPTION] horizontal resolution as the maximum degree of spherical harmonics"
     trunc::Int = DEFAULT_TRUNC
 
+    "[DERIVED] spectral spectrum"
+    spectrum::SP = Spectrum(trunc+2, trunc+1)
```

Comment:
@milankl I now added the `Spectrum` here in the `SpectralGrid` to have this as its "main location" that the other `LowerTriangularArray` then point to (didn't check yet that all of these are really references and not copies the `Spectrum`). Is this what you were thinking about with #732 as well?

Reply:
That's what I thought yes. But then `architecture` should also be taken as argument! The indices should be computeed at creation

Reply:
I'd do the same for `grid` and then use those as field constructors `field = zeros(grid, ...)`, `lta = zeros(spectrum, ...)` I guess both should then be considered only 2D so that we'd need to supply the vertical dimension somhow, e.g. `field = zeros(grid, nlayers)`